### PR TITLE
reviewed the docs functions to fix broken links and other additions

### PR DESCRIPTION
**Pull request by hugovalente-pm**
…le function list and callout when error on Cloud appears

##### Summary

Reviewed the docs functions to:
* fix broken links
* add current available functions
* callout for when user gets an error on Cloud

##### Test Plan

N/A

##### Additional Information

N/A

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

File: docs/cloud/netdata-functions.md [hunk 1]
```diff
@@ -23,9 +23,18 @@ A function is a  `key`  -  `value`  pair. The  `key`  uniquely identifies the fu
 function (i.e. code) to be run by a data collector when
 the function is invoked.
 
-For more details please check out documentation on our first collector that exposes
-functions - [plugins.d](/docs/nightly/references/collectors-references/plugins.d/#function)
+For more details please check out documentation on how we use our internal collector to get this from the first collector that exposes
+functions - [plugins.d](https://github.com/netdata/netdata/blob/master/docs/nightly/references/collectors-references/plugins.d/#function).
```
Comment (tkatsoulas) [inline docs/cloud/netdata-functions.md:27]:
```suggestion
functions - [plugins.d](https://github.com/netdata/netdata/tree/master/collectors/plugins.d#function).
```

File: docs/cloud/netdata-functions.md [hunk 2]
```diff
@@ -41,16 +50,16 @@ concern is to ensure no sensitive
 information or disruptive actions are exposed through the Agent's API.
 
 With the communication between the Netdata Agent and Netdata Cloud being
-through [ACLK](https://github.com/netdata/netdata/blob/master/docs/concepts/netdata-agent/aclk.md#ACLK) this
+through [ACLK](https://github.com/netdata/netdata/blob/master/docs/nightly/setup/agent-cloud-link-aclk) this
```
Comment (tkatsoulas) [inline docs/cloud/netdata-functions.md:53]:
```suggestion
through [ACLK](https://github.com/netdata/netdata/blob/master/aclk/README.md) this
```

File: docs/cloud/netdata-functions.md [hunk 3]
```diff
@@ -41,16 +50,16 @@ concern is to ensure no sensitive
...
 concern is addressed.
 
 ## Related Topics
 
 ### **Related Concepts**
 
-- [ACLK](https://github.com/netdata/netdata/blob/master/docs/concepts/netdata-agent/aclk.md)
-- [plugins.d](https://github.com/netdata/netdata/tree/master/collectors/plugins.d)
+- [ACLK](https://github.com/netdata/netdata/blob/master/docs/nightly/setup/agent-cloud-link-aclk)
```
Comment (tkatsoulas) [inline docs/cloud/netdata-functions.md:60]:
```suggestion
- [ACLK](https://github.com/netdata/netdata/blob/master/aclk/README.md)
```

Comment (tkatsoulas) [inline docs/cloud/netdata-functions.md:60]:
```suggestion
- [ACLK](https://github.com/netdata/netdata/blob/master/aclk/README.md)
```

**Comment by hugovalente-pm** [comment]
@tkatsoulas converting to draft since the links aren't working as expected, can you advice?

https://user-images.githubusercontent.com/82235632/214861457-d9920399-2ed1-43f0-a2ac-682c604b0500.mp4

what is expected

https://user-images.githubusercontent.com/82235632/214861490-3083f389-1bf1-4a5e-b649-852509a8643b.mp4

**Comment by hugovalente-pm** [comment]
discussed on [this ](https://github.com/netdata/netdata/pull/14331#issuecomment-1405084147) with @tkatsoulas and the full links to the GH file is the way to go.
moving the PR back to "normal" state

File: docs/cloud/runtime-troubleshooting-with-functions.md [hunk 4]
```diff
@@ -16,8 +16,8 @@ These routines can be used to retrieve additional information to help you troubl
 ### Prerequisites
 
 The following is required to be able to run Functions from Netdata Cloud.
-* Have at least one capable node claimed to one of your Spaces (node on a version higher than `v1.37`)
-* Ensure that node has the collector that exposes the function enabled
+* Have at least one capable node claimed to one of your Spaces where node needs to be on a version higher than `v1.37`
```
Comment (shyamvalsan) [inline docs/cloud/runtime-troubleshooting-with-functions.md:19]:
```suggestion
* At least one of the nodes claimed to your Space should be on a Netdata agent version higher than `v1.371.
```

File: docs/cloud/runtime-troubleshooting-with-functions.md [hunk 5]
```diff
@@ -16,8 +16,8 @@ These routines can be used to retrieve additional information to help you troubl
...
+* Ensure that node has the collector that exposes the function you want enabled ([see current available functions](https://github.com/netdata/netdata/blob/master/docs/cloud/netdata-functions.md#what-are-the-current-available-functions)
```
Comment (shyamvalsan) [inline docs/cloud/runtime-troubleshooting-with-functions.md:20]:
```suggestion
* Ensure that the node has the collector that exposes the function you want enabled ([see current available functions](https://github.com/netdata/netdata/blob/master/docs/cloud/netdata-functions.md#what-are-the-current-available-functions)
```

File: docs/cloud/netdata-functions.md [hunk 6]
```diff
@@ -23,9 +23,18 @@ A function is a  `key`  -  `value`  pair. The  `key`  uniquely identifies the fu
 function (i.e. code) to be run by a data collector when
 the function is invoked.
 
-For more details please check out documentation on our first collector that exposes
-functions - [plugins.d](/docs/nightly/references/collectors-references/plugins.d/#function)
+For more details please check out documentation on how we use our internal collector to get this from the first collector that exposes
+functions - [plugins.d](https://github.com/netdata/netdata/tree/master/collectors/plugins.d#function).
 
+#### What are the current available functions?
```
Comment (shyamvalsan) [inline docs/cloud/netdata-functions.md:29]:
```suggestion
#### What functions are currently available?
```

File: docs/cloud/runtime-troubleshooting-with-functions.md (hunk 4 above)
Comment (ilyam8) [reply docs/cloud/runtime-troubleshooting-with-functions.md:19]: a typo v1.371 => v1.37.1

Comment (hugovalente-pm) [reply docs/cloud/runtime-troubleshooting-with-functions.md:19]: I'll take in @shyamvalsan 's suggestion and will take care of the typo. @ilyam8 
thanks for checking that, I thought the functions capability had been introduced on `v1.37`

Comment (hugovalente-pm) [reply docs/cloud/runtime-troubleshooting-with-functions.md]: pushed a commit for this

**Review by shyamvalsan — APPROVED** [review]

File: docs/cloud/netdata-functions.md [hunk 7]
```diff
@@ -41,16 +50,16 @@ concern is to ensure no sensitive
...
 concern is addressed.
 
 ## Related Topics
 
 ### **Related Concepts**
 
-- [ACLK](https://github.com/netdata/netdata/blob/master/docs/concepts/netdata-agent/aclk.md)
-- [plugins.d](https://github.com/netdata/netdata/tree/master/collectors/plugins.d)
+- [ACLK](https://github.com/netdata/netdata/blob/master/aclk/README.md)
+- [plugins.d]((https://github.com/netdata/netdata/blob/master/docs/nightly/references/collectors-references/plugins.d)
```
Comment (ilyam8) [inline docs/cloud/netdata-functions.md:61]:
```suggestion
- [plugins.d](https://github.com/netdata/netdata/blob/master/docs/nightly/references/collectors-references/plugins.d)
```

File: docs/cloud/runtime-troubleshooting-with-functions.md [hunk 8]
```diff
@@ -32,10 +32,12 @@ The following is required to be able to run Functions from Netdata Cloud.
...
+
 ## Related Topics
 
 ### **Related Concepts**
-- [Netdata Functions](https://github.com/netdata/netdata/blob/master/docs/concepts/guided-troubleshooting/netdata-functions.md)
+- [Netdata Functions](https://github.com/netdata/netdata/blob/master/docs/cloud/netdata-functions.md)
 
 #### Related References documentation
-- [External plugins overview](https://github.com/netdata/netdata/tree/master/collectors/plugins.d#function)
+- [External plugins overview](https://github.com/netdata/netdata/blob/master/docs/nightly/references/collectors-references/plugins.d/#function)
```
Comment (ilyam8) [inline docs/cloud/runtime-troubleshooting-with-functions.md:43]: This link is 404

**Review by tkatsoulas — APPROVED** [review]

File: docs/cloud/netdata-functions.md (hunk 7 above)
Comment (tkatsoulas) [inline docs/cloud/netdata-functions.md:61]:
```suggestion
- [plugins.d](https://github.com/netdata/netdata/tree/master/collectors/plugins.d#function)
```

File: docs/cloud/runtime-troubleshooting-with-functions.md (hunk 8 above)
Comment (tkatsoulas) [inline docs/cloud/runtime-troubleshooting-with-functions.md:43]:
```suggestion
- [External plugins overview](https://github.com/netdata/netdata/tree/master/collectors/plugins.d#function)
```

Comment (hugovalente-pm) [reply docs/cloud/runtime-troubleshooting-with-functions.md]: seems @tkatsoulas made a suggestion will try that one

Comment (hugovalente-pm) [reply docs/cloud/runtime-troubleshooting-with-functions.md]: fixed by putting the direction to the README.md file

**Review by cakrit — DISMISSED** [review]

File: docs/cloud/netdata-functions.md [hunk 9]
```diff
@@ -41,16 +50,16 @@ concern is to ensure no sensitive
...
 
-- [ACLK](https://github.com/netdata/netdata/blob/master/docs/concepts/netdata-agent/aclk.md)
-- [plugins.d](https://github.com/netdata/netdata/tree/master/collectors/plugins.d)
+- [ACLK](https://github.com/netdata/netdata/blob/master/aclk/README.md)
+- [plugins.d](https://github.com/netdata/netdata/blob/master/collectors/plugins.d/README.md)
 
 ### Related Tasks
 
-- [Run-time troubleshooting with Functions](docs/nightly/tasks/operations/runtime-troubleshootting-with-function)
+- [Run-time troubleshooting with Functions](https://github.com/netdata/netdata/blob/master/docs/nightly/operations/run-time-troubleshooting-with-functions.md)
```
Comment (ilyam8) [inline docs/cloud/netdata-functions.md:65]: This link is 404. Is it expected?

Comment (hugovalente-pm) [reply docs/cloud/netdata-functions.md:65]: @ilyam8 I tested this after running the ingest flow locally and launching learn and this is working

https://user-images.githubusercontent.com/82235632/215470570-e9348bcc-7662-470d-8a0b-d47f9e9ded25.mp4

Comment (tkatsoulas) [inline docs/cloud/netdata-functions.md:65]:
```suggestion
- [Run-time troubleshooting with Functions](https://github.com/netdata/netdata/blob/master/docs/cloud/runtime-troubleshooting-with-functions.md_
```

Comment (hugovalente-pm) [reply docs/cloud/netdata-functions.md:65]: seems this was referencing some existing file, from a previous merge, that is also getting updated 🤷 
I'm committing a fix

Comment (hugovalente-pm) [reply docs/cloud/netdata-functions.md]: https://github.com/netdata/netdata/pull/14331/commits/49570927c96a9451087c2627983e53258a820738

**Review by tkatsoulas — APPROVED** [review]

**Pull request merged**